### DISCUSSION TITLE
If an EventSubscriber experiences a TimesheetCreatePostEvent, how do I redirect it?

### DISCUSSION THREAD

**Discussion by seleondevel** [kimai/kimai#4883] [Plugins]
When registering, modifying, and deleting timesheet, I would like to call the Google Calendar API to register, modify, and delete the schedule in the Google calendar.

If an EventSubscriber experiences a TimesheetCreatePostEvent, how do I redirect it?

    public function onTimesheetCreatePostEvent(TimesheetCreatePostEvent $event){
        $this->logger->info(sprintf('TimesheetCreatePostEvent!!'));
        return $this->redirectToRoute('google_login');
}

//#[Route(path: '/google_login')]
    #[Route(path: '/google_login', name: 'google_login', methods: ['GET', 'POST'])]
    public function homepage4(): Response
    {        
        $this->logger->info(sprintf('google_login!!'));

        return new Response('google oauth2 success');
}

Even if I call redirectToRoute, I can't get any response, what should I do?

**Comment by seleondevel** [comment]
Or, when registering timesheet in a docline postPersist event, should I call redirectToRoute?

**Reply by kevinpapst** [reply]
The Event System is a core component of Kimai. You are better of reading their docs. 
To me that sounds as if you should call the Google APIs to achieve that, instead of using a redirect.

**Comment by seleondevel** [comment]
public function onTimesheetCreatePostEvent(TimesheetCreatePostEvent $event){
    $this->logger->info(sprintf('TimesheetCreatePostEvent!!'));
} 
    
public function postPersist(Timesheet $timesheet, postPersistEventArgs $eventArgs)
{   
    $this->logger->info(sprintf('Timesheet create postPersist!'));
}
    
How do I call the google calendar API from the function above?

**Reply by seleondevel** [reply]
@kevinpapst Can't you give me some advice? What should I look for?

**Reply by kevinpapst** [reply]
I develop Kimai and give it to everyone for free. That's it.
If you want to pay for support, I am all yours.

If you need help with Symfony related questions (all of the above are) then there is the Symfony documentation, their Slack workspace, and various online platforms to ask questions.
